### DISCUSSION TITLE
Unknown magic byte during deserialization

### DISCUSSION THREAD

**Discussion by sigalite** [Apicurio/apicurio-registry#1987] [Q&A]
Hi
in AbstractKafkaDeserializer in the deserialize there is a validation regarding  Unknown magic byte!.
we saw that in the serializer area this wrapper is added only in case here is an no header. but in the desriliazer this validation is always done regardless to the header existance.
can u please clarify why this is. the case?

**Comment by EricWittmann** [comment]
@famartinrh do we check for the magic byte in the deserializer **after** checking for the existence of a header or before?

**Reply by famarting** [reply]
https://github.com/Apicurio/apicurio-registry/blob/master/serdes/serde-common/src/main/java/io/apicurio/registry/serde/AbstractKafkaDeserializer.java

deserializer uses identifiers from headers, if headers are present, or read identifier from body, if magic byte is detected. that is done in that order.

I don't see why you say "the validation is always done regardless to the header existence" in the deserializer. That's not true https://github.com/Apicurio/apicurio-registry/blob/master/serdes/serde-common/src/main/java/io/apicurio/registry/serde/AbstractKafkaDeserializer.java#L117 .

Anyway, the idea of checking all the possibilities in the deserializer is to be able to read and successfully deserialize all messages that the kafka consumer reads, regardless of the place where the identifiers are located.

**Reply by EricWittmann** [reply]
Thanks for the answer @famartinrh - that's what I thought.  Does this help @sigalite ?

**Comment by sigalite** [comment]
Yes. Thanks for the clarification.

Regards
Sigalit
